transportable-udfs-trino-plugin/src/main/java/com/linkedin/transport/trino/TransportConnector.java

```diff
@@ -60,11 +58,16 @@ public TransportConnector(ConnectorMetadata connectorMetadata, FunctionProvider
   @Inject
   public TransportConnector(TransportConfig config) {
     ClassLoader classLoaderForFactory = TransportConnectorFactory.class.getClassLoader();
-    List<URL> jarUrlList = getUDFJarUrls(config);
-    log.info("The URLs of Transport UDF jars: " + jarUrlList);
-    TransportUDFClassLoader classLoaderForUdf = new TransportUDFClassLoader(classLoaderForFactory, jarUrlList);
-    ServiceLoader<StdUdfWrapper> serviceLoader = ServiceLoader.load(StdUdfWrapper.class, classLoaderForUdf);
-    List<StdUdfWrapper> stdUdfWrappers = ImmutableList.copyOf(serviceLoader);
+    List<List<URL>> allJarUrlList = getUDFJarUrls(config);
+    log.info("The URLs of Transport UDF jars: " + allJarUrlList);
+    List<StdUdfWrapper> stdUdfWrappers = new ArrayList<>();
+    // create one TransportUDFClassLoader to load jars for one sub directory under Transport UDF repo
+    for (List<URL> jarUrlList : allJarUrlList) {
+      TransportUDFClassLoader classLoaderForUdf = new TransportUDFClassLoader(classLoaderForFactory, jarUrlList);
```

Comment:
Since we may not be able to verify the classloader during testing. Can we add a log statement to print the classloader info and the matching group of `StdUdfWrapper`?  
Some thing like the following.
```
Creating TransportUDFClassLoader <> ...
-- function1
-- function2
...
```

Reply:
why can't we verify the classloader during testing?

Reply:
I discussed with @yiqiangin we did not have a simple way to get the classloader after the plugins are loaded. The functions are represented as `StdUdfWrapper` in the connector. Even if we create a testing transport plugin/connector that makes them public, it does make the UDF specific classloader accessible. @xkrogen Can you shed some light on this?
I think this can be a future improvement for robustness for Transport, so that we can expedite our upgrade progress. We can test if the functions are behaving correctly with a local setup.

Reply:
So we can pretty easily test this by asking `TransportConnector` to load the two local UDF JARs you've configured and validating that the classloaders all look like we expect. This works:

```java
  @Test
  public void test() {
    String udfRepoDir = getClass().getClassLoader().getResource(TRANSPORT_UDF_REPO_DIR).getPath();
    TransportConfig config = new TransportConfig();
    config.setTransportUdfRepo(udfRepoDir);
    TransportConnector connector = new TransportConnector(config);
    TransportFunctionProvider fnProvider = (TransportFunctionProvider) connector.getFunctionProvider().get();

    ClassLoader parentStdUdfWrapperClassloader = StdUdfWrapper.class.getClassLoader();
    Set<ClassLoader> classLoaders = new HashSet<>();
    for (StdUdfWrapper udfWrapper : fnProvider.functions.values()) {
      classLoaders.add(udfWrapper.getClass().getClassLoader());
      assertTrue(udfWrapper.getClass().getClassLoader() instanceof TransportUDFClassLoader);
      // the classloader of the superclass (StdUdfWrapper) should always be the same parent loader
      assertEquals(udfWrapper.getClass().getSuperclass().getClassLoader(), parentStdUdfWrapperClassloader);
    }
    // two UDF JARs are being loaded, so we expect two classloaders
    assertEquals(classLoaders.size(), 2);
  }
```

(I had to play around with the visibility of a few fields to get everything exposed; probably should check on how it can be done most cleanly)

Reply:
Done. Thanks.